File: srclib/v3/spss/srclib/execute.py
# execute function

import srclib
import os, string, subprocess, sys

def execute(prog, path, mode, args):
  cmd = []  # execute srchset
  cmd.append(os.path.normpath(os.path.join(srclib.__path__[0], "..", "..", "bin", prog)))
  cmd.append(path + ".set")
  cmd.append("/spss")
  if mode:
    cmd.append("/debug")
  if args:
    cmd.extend(args)
  if sys.platform.startswith("win"):  # windows
    rc = subprocess.call(cmd)
  else:  # linux
    for i in range(len(cmd)):
      if cmd[i].find(" ") >= 0:
        cmd[i] = "'{0}'".format(cmd[i])
    cmd = string.join(cmd)
    rc = os.system(cmd)
    rc = 0
  return rc

File: srclib/v3/R/execute.R
# execute function

execute <- function(prog="", args="") {
  cmd <- file.path(srclib, "..", "bin", prog, fsep=.Platform$file.sep)
  if (regexpr(" ", srclib, fixed=TRUE) >= 0) 
    cmd <- dQuote(cmd)
  if (regexpr(" ", path, fixed=TRUE) < 0) cmd <- paste(cmd, paste(path, ".set", sep=""), sep=" ")
  else cmd <- paste(cmd, dQuote(paste(path, ".set", sep="")), sep=" ")
  cmd <- paste(cmd, "/r", sep=" ")
  if (!identical(test, "")) cmd <- paste(cmd, "/debug", sep=" ")
  if (!identical(args, "")) cmd <- paste(cmd, args, sep=" ")
  if (system(cmd) != 0)
    msg <<- paste("Abnormal termination of ", prog)
}

